src/base/rss/rssdownloadrule.cpp

```diff
@@ -42,33 +42,43 @@
 
 using namespace Rss;
 
-const QString episodeRegex = "[^A-Za-z0-9]((?:S\\d{1,}(?:E\\d{1,})+)|(?:\\d{1,}x\\d{1,})|(?:\\d{4}\\.\\d{2}\\.\\d{2}))[^A-Za-z0-9]";
+static const QString episodeRegex = "(?:^|[^A-Za-z0-9])((?:S\\d{1,}(?:E\\d{1,})+)|(?:\\d{1,}x\\d{1,})|(?:\\d{4}\\.\\d{2}\\.\\d{2}))(?:[^A-Za-z0-9]|$)";
 
-QString computeEpisodeName(QString matchedEpisode)
+QString computeEpisodeName(const QString &article)
 {
+    QRegExp f(episodeRegex, Qt::CaseInsensitive);
+    // See if we can extract an season/episode number or date from the title
+    if (f.indexIn(article) == -1) {
+        return QString();
+    }
+
+    // Extract out the episode identifier
+    QString matchedEpisode = f.cap(1);
+
+    // Is it in the S01E01 format?
     if (matchedEpisode.startsWith('S')) {
         QRegExp r("S(\\d+)E(\\d+)");
         if (r.indexIn(matchedEpisode) != -1) {
-            return QString("%1x%2").arg(r.cap(1).toInt())
-                                   .arg(r.cap(2).toInt());
+            return QString("%1x%2").arg(r.cap(1).toInt()).arg(r.cap(2).toInt());
         }
     }
+    // Is it in the 01x01 format?
     else if (matchedEpisode.contains('x')) {
         QRegExp r("(\\d+)x(\\d+)");
```

Comment:
I don't use RSS rules and have no idea about common episode naming rules, but here are my two cents: wouldn't it be simpler to convert the article name to uppercase before matching and keep regexps simpler?